src/utils/atomWithStorage.ts

```diff
@@ -75,6 +75,17 @@ export function createJSONStorage<Value>(
     setItem: (key, newValue) =>
       getStringStorage().setItem(key, JSON.stringify(newValue)),
     removeItem: (key) => getStringStorage().removeItem?.(key), // TODO remove optional chaining
+    subscribe: (key, callback) => {
+      const storageEventCallback = (e: StorageEvent) => {
+        if (e.key === key && e.newValue) {
+          callback(JSON.parse(e.newValue))
+        }
+      }
+      window.addEventListener('storage', storageEventCallback)
+      return () => {
+        window.removeEventListener('storage', storageEventCallback)
+      }
+    },
```

Comment:
`createJSONStorage` is an exported function and used with other storages like AsyncStorage in RN.

We need to apply this logic only for `defaultStorage`.